### PR TITLE
changed vvm to pysolcx

### PR DESCRIPTION
**Pull request by sabotagebeats**
This is a draft and requires review

note: this code was copied and modified from https://github.com/ApeWorX/ape-vyper/blob/main/ape_vyper/compiler.py

### CODE REVIEW

**Comment by sabotagebeats** [comment]
`WARNING: No project files detected in '/home/somatt/work/ape/token/contracts'`

**Comment by sabotagebeats** [comment]
`WARNING: No project files detected in '~/ape/token/contracts'`

**Review by fubuloubu — APPROVED** [review]

**Comment by sabotagebeats** [comment]
need to update workflow yaml file - doesn't have module updated, h/t @fubuloubu

**Comment by sabotagebeats** [comment]
need to add testing similarly to ape-vyper h/t @fubuloubu

File: ape_solidity/compiler.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                "compact-format",
+                "devdoc",
+                "hashes",
+                "interface",
+                "metadata",
+                "opcodes",
+                "srcmap",
+                "srcmap-runtime",
+                "storage-layout",
+                "userdoc",
```
Comment (fubuloubu) [inline ape_solidity/compiler.py:90]: We only use 5 of these keys below

**Review by fubuloubu — APPROVED** [review]

**Comment by sabotagebeats** [comment]
error 
```
>       progress_bar = tqdm(total=total_size, unit="iB", unit_scale=True)
E       TypeError: 'NoneType' object is not callable
```

File: ape_solidity/compiler.py (hunk 1 above)
Comment (sabotagebeats) [reply ape_solidity/compiler.py:90]: @fubuloubu according to the documentation it says that 
```
output_values List
Compiler outputs to return. Valid options depend on the version of solc. If not given, all possible outputs for the active version are returned.
```
how can I tell which of the keys we need?

**Comment by sabotagebeats** [comment]
``` 
Run flake8 ./<MODULE_NAME> ./tests ./setup.py
  flake8 ./<MODULE_NAME> ./tests ./setup.py
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.8.10/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.10/x64/lib
/home/runner/work/_temp/0e8cef5f-3d75-4fda-bb45-bc0decf783a5.sh: line 1: MODULE_NAME: No such file or directory
Error: Process completed with exit code 1.
```

it looks like `MODULE_NAME` is not set

**Comment by sabotagebeats** [comment]
https://github.com/ApeWorX/ape-solidity/blob/755c50de78647a5ad7ebe82a6b4ab6df01216909/ape_solidity/compiler.py#L92

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

ape_solidity/compiler.py:92: error: Item "None" of "Optional[Any]" has no attribute "select"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by sabotagebeats** [comment]
> need to update workflow yaml file - doesn't have module updated, h/t @fubuloubu

https://github.com/ApeWorX/ape-solidity/blob/755c50de78647a5ad7ebe82a6b4ab6df01216909/.github/workflows/publish.yaml#L30

**Comment by sabotagebeats** [comment]
hook into github template system

**Review by sabotagebeats — COMMENTED** [review]
if not none

**Comment by ShadeUndertree** [comment]
Fixes https://github.com/ApeWorX/ape/issues/69

File: tests/contracts/solcontract.sol [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+function foo() pure returns(bool) {
```
Comment (sabotagebeats) [inline tests/contracts/solcontract.sol:5]: wrap this in a contract, named `solcontract`

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**